lib/spack/spack/test/spec_semantics.py

```diff
@@ -985,6 +985,81 @@ def test_forwarding_of_architecture_attributes(self):
         assert 'avx512' not in spec.target
         assert spec.target < 'broadwell'
 
+    @pytest.mark.parametrize('transitive', [True, False])
+    def test_splice(self, transitive):
+        # Tests the new splice function in Spec using a somewhat simple case
+        # with a variant with a conditional dependency.
+        # TODO: Test being able to splice in different provider for a virtual.
+        # Example: mvapich for mpich.
+        spec = Spec('splice-t')
+        dep = Spec('splice-h+foo')
+        spec.concretize()
+        dep.concretize()
+        # Sanity checking that these are not the same thing.
+        assert dep.dag_hash() != spec['splice-h'].dag_hash()
+        assert dep.build_hash() != spec['splice-h'].build_hash()
+        # Do the splice.
+        out = spec.splice(dep, transitive)
+        # Returned spec should still be concrete.
+        assert out.concrete
+        # Traverse the spec and assert that the targets are correct.
```

Comment:
Targets are a term of art in Spack, so we should avoid using the term here when we aren't referring to architecture.

```suggestion
        # Traverse the spec and assert that all dependencies are accounted for
```

Reply:
Applied in latest commit.